Automattic/pocket-casts-android
# Search history - Analytics

Part of #752 

## Description
This adds analytics events for search history and search shown/ dismissed events. (p1676369395793819-slack-C03JCLLR9C3)

## Testing Instructions

<details><summary>Tap search history item </summary>

- Build a search history of search terms, podcasts, and folder
- Tap on a search term row from the search history
- ✅ Notice `Tracked: search_history_item_tapped, Properties: {"source": SOURCE ,"type":"search_term", ]`  in logs
   where `SOURCE` is source from where search is opened (podcasts_list or discover)
- Go back to search history 
- Tap on a podcast row from the search history
- ✅ Notice `Tracked: search_history_item_tapped, Properties: {"source": SOURCE ,"type":"podcast", "uuid": UUID ]`  in logs
   where `SOURCE` is source from where the search is opened (`podcasts_list` or `discover`) and `UUID` is podcast uuid
- Go back to search history 
- Tap on a folder row from the search history
- ✅ Notice `Tracked: search_history_item_tapped, Properties: {"source": SOURCE ,"type":"folder", "uuid": UUID ]`
   where `SOURCE` is source from where the search is opened (`podcasts_list` or `discover`) and `UUID` is folder uuid
</details>

<details><summary>Delete search history item </summary>

- Build a search history of search terms, podcasts, and folder
- Go to search history and tap the x button on a search term row
- ✅ Notice `Tracked: search_history_item_delete_button_tapped, Properties: {"source": SOURCE ,"type":"search_term", ]`  in logs
   where `SOURCE` is source from where search is opened (podcasts_list or discover)
- Go back to search history 
- Tap the x button on a podcast search history row
- ✅ Notice `Tracked: search_history_item_delete_button_tapped, Properties: {"source": SOURCE ,"type":"podcast", "uuid": UUID ]`  in logs
   where `SOURCE` is source from where the search is opened (`podcasts_list` or `discover`) and `UUID` is podcast uuid
- Go back to search history 
- Tap the x button on a folder search history row
- ✅ Notice `Tracked: search_history_item_delete_button_tapped, Properties: {"source": SOURCE ,"type":"folder", "uuid": UUID ]`  in logs
   where `SOURCE` is source from where the search is opened (`podcasts_list` or `discover`) and `UUID` is folder uuid
</details>

<details><summary>Clear all</summary>

- Build a search history of less than 3 items
- Tap clear all above search history list
- ✅ Notice `Tracked: search_history_cleared, Properties: {"source":SOURCE", ]`  in logs
where `SOURCE` is source from where the search is opened
- Build a search history of more than 3 items
- Tap clear all above search history list followed by "Clear Search History" in the confirmation dialog
- ✅ Notice `Tracked: search_history_cleared, Properties: {"source":SOURCE", ]`  in logs
   where `SOURCE` is source from where the search is opened
</details>


<details><summary>Search shown/ dismissed</summary>

- Go to podcasts tab
- Tap search bar
- ✅ Notice `Tracked: search_shown, Properties: {"source": podcasts_list, ]` in logs
- Tap on the back arrow
- ✅ Notice `Tracked: search_dismissed, Properties: {"source": podcasts_list, ]` in logs
- Reopen the screen and navigate back using back gesture
- ✅ Notice `Tracked: search_dismissed, Properties: {"source": podcasts_list, ]` in logs

Repeat the above steps for search from "discover" tab and from "onboarding recommendations". Notice that the source changes to "discover" and "onboarding_recommendations" respectively.

</details>
